cupy/core/carray.cuh

```diff
@@ -259,30 +259,27 @@ public:
   }
 
   __device__ void set(ptrdiff_t i) {
-    if (ndim == 0) {
+    if (ndim == 1) {
```

Comment:
I realized `ndim == 0` is exluded by the template only after seeing this change.
I think it should be mentioned as a comment here.